tbx/core/blocks.py

```diff
@@ -622,9 +626,18 @@ def clean(self, value):
         struct_value = super().clean(value)
 
         errors = {}
+        non_block_errors = ErrorList()
```

Comment:
**Issue**: `non_block_errors` are applied on the PhotoCollageBlock as a whole, rather than on individual blocks within in ([ref](https://docs.wagtail.org/en/latest/advanced_topics/streamfield_validation.html#controlling-where-error-messages-are-rendered))

This is what happens:

<details>
  <summary>Screenshot</summary>
  
![Screenshot from 2024-04-17 18-57-05](https://github.com/torchbox/torchbox.com/assets/7713776/3de8a14c-85cf-4ab9-9f8f-c28975de6802)
  
</details>

**Suggestion**: `block_errors` should suffice, as follows:

```py
    def clean(self, value):
        struct_value = super().clean(value)

        errors = {}
        button_link = value.get("button_link")
        button_text = value.get("button_text")

        has_button = bool(button_link and button_text)
        has_intro = bool(value.get("intro"))

        if (has_intro or has_button) and not value["title"]:
            error = ErrorList(
                [
                    ValidationError(
                        "You cannot add a button or intro without also adding a title"
                    )
                ]
            )
            errors["title"] = error

        if button_link and not button_text:
            error = ErrorList(
                [ValidationError("You must add button text for the button link.")]
            )
            errors["button_text"] = error

        if button_text and not button_link:
            error = ErrorList(
                [
                    ValidationError(
                        "You must specify a button link above, for the button text you have provided."
                    )
                ]
            )
            errors["button_text"] = error

        if errors:
            raise blocks.StructBlockValidationError(block_errors=errors)
        return struct_value
``` 

With the above suggestion, this is what it looks like:

<details>
  <summary>Screenshot</summary>
    
![image](https://github.com/torchbox/torchbox.com/assets/7713776/2655b4b9-8ccf-4e21-9f55-c3a160b88cc9)
  
</details>

Reply:
Implemented! thank you